cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/AbstractStreamService.java

```diff
@@ -97,4 +112,39 @@ protected ScheduledExecutorService executor() {
     executor = Executors.newSingleThreadScheduledExecutor(Threads.createDaemonThreadFactory("heartbeats-scheduler"));
     return executor;
   }
+
+  /**
+   * Get the size of events ingested by a stream since its creation, in bytes.
+   * @param streamId id of the stream
+   * @return Size of events ingested by a stream since its creation
+   * @throws IOException when getting an error retrieving the metric
+   */
+  protected long getStreamEventsSize(Id.Stream streamId) throws IOException {
+    MetricDataQuery metricDataQuery = new MetricDataQuery(
```

Comment:
No it's still the case, the metric can still go down according to @gsps1, but we had the same problem with the previous logic in which we were reading the files of the stream directly, because the stream file janitor could suddenly get rid of some data in a file.

Sasha told me not to worry about deleting metrics, because they planned on never documenting it.